PYME/Acquire/protocol.py

```diff
@@ -214,7 +214,7 @@ def Init(self, spooler):
 
     def OnFrame(self, frameNum):
         if frameNum > self.startFrame:
-            fn = int(floor((frameNum - self.startFrame)/self.dwellTime) % len(self.zPoss))
```

Comment:
I'm pretty sure this should be floor - what is your reasoning to the contrary?

Reply:
Protocol logic is that a protocol task executes strictly **after** the designated frame

Reply:
Correct, it should be a floor and it *is* a floor, just don't need the floor call. Int truncates, i.e. rounds towards zero. Floor rounds towards negative inf, but since the line above ensures we only get here if we're positive, int is both our int-cast and our floor.

Reply:
integer truncation / rounding behaviour is somewhat platform specific, and the floor makes this explicit and the code much more readable.

Reply:
We are not rounding. Name an OS that "truncates" a positive float to the next highest int?

Do we not call this on every frame? ProtocolFocus events are fired after this line and I can run my camera at kHz. I want this to be as fast as we can make it.

Reply:
truncation/rounding/cast behaviour depends on the processor, flags, and compiler, not platform. In any case int(xxx) truncation is not guaranteed, not clear, and with `floor()` taking around 1us (i.e. 3 orders of magnitude faster than anything we need to care about) this is not a case where we should be optimizing.

Reply:
I suspect the line showed up slightly red in the profile, but this is likely a case of being caught out by profiler overhead - there are a bunch of function calls on this line which all incur significant profiling overhead.